File: api/src/main/java/org/snomed/snap2snomed/controller/dto/ValidationResult.java
package org.snomed.snap2snomed.controller.dto;

import lombok.Data;

import java.util.Set;

@Data
public class ValidationResult {

    Integer activeCount;
    Set<String> inactive;
    Set<String> absent;
    Set<String> invalid;

    public ValidationResult(Integer activeCount, Set<String> inactive, Set<String> absent, Set<String> invalid) {
        this.activeCount = activeCount;
        this.inactive = inactive;
        this.absent = absent;
        this.invalid = invalid;
    }

    @Override
    public String toString() {
        return "ValidationResult{" +
                "activeCount=" + activeCount +
                ", inactive=[" + inactive.size() + "]" +
                ", absent=[" + absent.size() + "]" +
                ", invalid=[" + invalid.size() + "]" +
                '}';
    }
}

File: ui/snapclient/src/app/_models/validation_result.ts

export class ValidationResult {
  get activeCount(): number {
    return this._activeCount;
  }

  set activeCount(value: number) {
    this._activeCount = value;
  }

  get inactive(): string[] {
    return this._inactive;
  }

  set inactive(value: string[]) {
    this._inactive = value;
  }

  get absent(): string[] {
    return this._absent;
  }

  set absent(value: string[]) {
    this._absent = value;
  }

  get invalid(): string[] {
    return this._invalid;
  }

  set invalid(value: string[]) {
    this._invalid = value;
  }

  private _activeCount: number = -1;
  private _inactive: string[] = [];
  private _absent: string[] = [];
  private _invalid: string[] = [];
}
